reader_concurrency_semaphore.hh

```diff
@@ -195,19 +197,23 @@ public:
 
     reader_concurrency_semaphore(int count,
             ssize_t memory,
+            sstring name,
             size_t max_queue_length = std::numeric_limits<size_t>::max(),
-            std::function<std::exception_ptr()> raise_queue_overloaded_exception = default_make_queue_overloaded_exception)
+            std::function<void()> prethrow_action = nullptr)
         : _resources(count, memory)
+        , _wait_list(expiry_handler(name))
+        , _name(std::move(name))
         , _max_queue_length(max_queue_length)
-        , _make_queue_overloaded_exception(raise_queue_overloaded_exception) {
-    }
+        , _prethrow_action(std::move(prethrow_action)) {}
 
     /// Create a semaphore with practically unlimited count and memory.
     ///
     /// And conversely, no queue limit either.
     explicit reader_concurrency_semaphore(no_limits)
-        : reader_concurrency_semaphore(std::numeric_limits<int>::max(), std::numeric_limits<ssize_t>::max()) {
-    }
+        : reader_concurrency_semaphore(
+                std::numeric_limits<int>::max(),
+                std::numeric_limits<ssize_t>::max(),
+                "unlimited RCS") {}
```

Comment:
I would have liked it better without the acronym, but I won't insist as I don't expect to ever actually see it.

Reply:
This is an all inclusive, 5-star pull request - acronym removed.

Reply:
Haha, thanks, I feel spoiled.